src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -110,23 +90,64 @@ protected JsonNode valueToJson(FieldValue value, Field field) {
     }
   }
 
-  private List<JsonNode> getJobResults(Job queryJob) {
+  public QueryResult getQueryResults(String queryId, int offset, int pageSize) {
+    final Job job = bigQuery.getJob(queryId);
+    if (job == null || !job.exists()) {
+      throw new RuntimeException("Unknown query " + queryId);
+    }
+    if (!job.isDone()) {
```

Comment:
I need to test the case where the query is ill-formed and make sure that errors are getting routed correctly.

Reply:
I've tested this and it works as expected; query errors are returned to the user from the new `query` endpoint. They are not reported during the initial call, which creates the job to run the query.